src/main/java/seedu/edulink/logic/commands/DeleteTagCommand.java

```diff
@@ -0,0 +1,98 @@
+package seedu.edulink.logic.commands;
+
+import static seedu.edulink.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.edulink.logic.parser.CliSyntax.PREFIX_ID;
+import static seedu.edulink.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import seedu.edulink.commons.util.ToStringBuilder;
+import seedu.edulink.logic.commands.exceptions.CommandException;
+import seedu.edulink.model.Model;
+import seedu.edulink.model.student.Id;
+import seedu.edulink.model.student.Student;
+import seedu.edulink.model.tag.Tag;
+
+/**
+ * Removes tags from a student.
+ */
+public class DeleteTagCommand extends Command {
+
+    public static final String COMMAND_WORD = "dtag";
+
+    public static final String MESSAGE_PERSON_NOTFOUND = "Can't find the person you specified.";
+    public static final String MESSAGE_DELETE_TAG_SUCCESS = "Deleted Tags: %1$s";
+    public static final String MESSAGE_USAGE = "Usage: " + COMMAND_WORD + " " + PREFIX_ID + "ID " + PREFIX_TAG + "Tag";
+    public static final String MESSAGE_TAG_NOT_FOUND = "Invalid Command"
+            + ", one or more tags you are looking for are not found.";
+
+    private final Id personToEditId;
+    private final Set<Tag> tags;
+
+    /**
+     * Creates a DeleteTagCommand to delete tags from a student.
+     *
+     * @param personToEditId the ID of the student user add tags to.
+     * @param tags           a set of tags that the user wish to delete from the student.
+     */
+    public DeleteTagCommand(Id personToEditId, Set<Tag> tags) {
+        requireAllNonNull(personToEditId, tags);
+
+        this.personToEditId = personToEditId;
+        this.tags = tags;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        List<Student> lastShownList = model.getFilteredPersonList();
+        Student studentToEdit = null;
```

Comment:
Can we avoid the  use of `null` here and instead try using `Optional` or just manipulate the `lastShownList` using Streams

Reply:
Thank you, issue resovled.